[CI] S3BlobStoreRepositoryTests.testSnapshotAndRestore fails in 7.3

Build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.3+multijob-unix-compatibility/os=fedora-29&&immutable/73/console

Reproduce:

```
./gradlew :plugins:repository-s3:testRepositoryCreds --tests "org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests.testSnapshotAndRestore" -Dtests.seed=26D87699878FD59B -Dtests.security.manager=true -Dtests.locale=sv -Dtests.timezone=Australia/Victoria -Dcompiler.java=12 -Druntime.java=8
```

Failure:

```
org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests > testSnapshotAndRestore FAILED
    java.lang.AssertionError: CloseIndexResponse failed - not acked
    Expected: <true>
         but: was <false>
```

Unfortunately the failure is not printed out by the test.

--- a/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java
@@ -131,6 +131,8 @@ public abstract class ESBlobStoreRepositoryIntegTestCase extends ESIntegTestCase
                 }
             }
 
+            // Wait for green so the close does not fail in the edge case of coinciding with a shard recovery that hasn't fully synced yet
+            ensureGreen();
             logger.info("-->  close indices {}", closeIndices);
             assertAcked(client().admin().indices().prepareClose(closeIndices.toArray(new String[closeIndices.size()])));
         }
@@ -195,6 +197,8 @@ public abstract class ESBlobStoreRepositoryIntegTestCase extends ESIntegTestCase
             int iterationToRestore = randomIntBetween(0, iterationCount - 1);
             logger.info("-->  performing restore of the iteration {}", iterationToRestore);
 
+            // Wait for green so the close does not fail in the edge case of coinciding with a shard recovery that hasn't fully synced yet
+            ensureGreen();
             logger.info("-->  close index");
             assertAcked(client().admin().indices().prepareClose(indexName));
 

